test/index.test.js

```diff
@@ -126,3 +126,23 @@ test("should handle empty source code", (t) => {
 	assert.strictEqual(transformSync(null).code, "");
 	assert.strictEqual(transformSync("").code, "");
 });
+
+test("should not polyfill using Symbol.Dispose", (t) => {
+	const inputCode = `
+	class Resource {
+		[Symbol.dispose]() { console.log("Disposed"); }
+	}
+	using r = new Resource();`;
```

Comment:
This fixture string seems like it will achieve the goal of detecting accidental downleveling of JS syntax.

My only comment is that it may be more verbose than needed, which might make the test brittle in future.  For example, whitespace changes might break it.  Such changes would be legitimate in `"transform"` mode because sourcemaps fix it up.

A less brittle fixture that could be used for all the cases in this PR and would probably still achieve the goal would simply be:

```suggestion
	using r = 0;`;
```

This is not a blocking issue, so I'm approving regardless.